src/NaCl.Core/Base/ChaCha20BaseIntrinsics.cs

```diff
@@ -0,0 +1,69 @@
+﻿#if INTRINSICS
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics.X86;
+using NaCl.Core.Base.ChaChaIntrinsics;
+
+namespace NaCl.Core.Base;
+
+public static class ChaCha20BaseIntrinsics
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static unsafe void ChaCha20(Span<uint> state, ReadOnlySpan<byte> input, Span<byte> output, ulong bytes)
+    {
+        if (!Sse3.IsSupported || !BitConverter.IsLittleEndian)
+            throw new Exception("Error this vectorisation is not supported on this CPU");
```

Comment:
We should avoid using the base `Exception` and instead use `NotSupportedException`, I would also suggest using a more specific error message like `The SSE3 vectorisation is not supported on the current CPU`.